compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/repl/ReplApi.kt

```diff
@@ -83,7 +84,9 @@ sealed class ReplCompileResult : Serializable {
                           val mainClassName: String,
                           val classes: List<CompiledClassData>,
                           val hasResult: Boolean,
-                          val classpathAddendum: List<File>) : ReplCompileResult()
+                          val classpathAddendum: List<File>) : ReplCompileResult() {
+        val type = LastInferredTypeHolder.inferredType.get()
```

Comment:
It would be easier to pass the expression type explicitly where `ReplCompileResult.CompiledClasses` is created at `GenericReplCompiler.compile` with:
```
val expression = psiFile.getChildOfType<KtExpression>()
compilerState.analyzerEngine.trace.bindingContext.getType(expression)
```
(+ handle the case when there's no expression in a REPL line)